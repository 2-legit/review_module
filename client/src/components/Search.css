.input-bar {
  border: solid silver 1px;
  border-radius: 4px;
  height: 30px;
  font-weight: bold;
}

.input-bar:focus {
  border: solid #008489 2px;
  border-radius: 4px;
  outline: none;
}

.search {
  position: relative;
}

.search input {
  text-indent: 30px
}

.search .fa-search {
  position: absolute;
  top: 8px;
  left: 7px;
  font-size: 15px;
}