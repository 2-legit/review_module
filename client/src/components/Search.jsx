import React from 'react';
import styles from './Search.css';
class Search extends React.Component {
  render() {
    return (
      <form className="search" onSubmit={this.props.searchSubmit}>
      <div>
        <i class="fas fa-search"></i>
        <input
        className="input-bar"
        type="search"
        onChange={this.props.searchBarTextChange}
        placeholder="Search reviews"></input>
      </div>
      </form>
    );
  }
}

export default Search;
