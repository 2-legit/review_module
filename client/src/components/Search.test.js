import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Search from './Search';

/* What do I want to test in Search that will prove it works?
  1. Changes the state on submission.
  2. Re-renders the page on state change.
  3. Can render the original state (original being allReviews) on reset button click.
  4. Will bold words in paragraphs? (this one may be tricky)
*/

describe('Search Test Suite', () => {
  test('Should have a matching snapshot', () => {
    const component = shallow(<Search />);
    expect(component).toMatchSnapshot();
  });
})