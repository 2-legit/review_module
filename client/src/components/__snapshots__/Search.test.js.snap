// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Test Suite Should have a matching snapshot 1`] = `ShallowWrapper {}`;
